presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -253,6 +254,8 @@ private synchronized void populateFromDbHelper(Map<Long, ResourceGroupSpecBuilde
         List<SelectorSpec> selectors = dao.getSelectors()
                 .stream()
                 .filter(selectorRecord -> resourceGroupIdTemplateMap.containsKey(selectorRecord.getResourceGroupId()))
```

Comment:
We are loading all selectors before filtering out ones don't apply to this cluster, which means that the load here grows as the number of cluster grows. This should be fine when we have a few dozen clusters. But it means that merging the `selectors` and `exact_match_selectors` will not happen because the other table can be huge.